source/common/init/target_impl.h

```diff
@@ -68,8 +68,9 @@ class TargetImpl : public Target, Logger::Loggable<Logger::Id::init> {
 
   /**
    * Signal to the init manager that this target has finished initializing. This is safe to call
-   * any time. Calling it before initialization begins or after initialization has already ended
-   * will have no effect.
+   * any time.
+   * Notes: Calling it before initialization begins is allowed.
```

Comment:
Should probably clarify what happens when you call `ready()` before the init manager has called the target.